### PR TITLE
PV2-2013 - Period End Metric Calculation Service

### PR DESCRIPTION
**Pull request by keenesolutions**

### CODE REVIEW

File: src/SFA.DAS.Payments.Monitoring.Metrics.Application/SFA.DAS.Payments.Monitoring.Metrics.Application.csproj [hunk 1]
```diff
@@ -5,13 +5,15 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <ProjectReference Include="..\SFA.DAS.Payments.Application\SFA.DAS.Payments.Application.csproj" />
-    <ProjectReference Include="..\SFA.DAS.Payments.Monitoring.Metrics.Data\SFA.DAS.Payments.Monitoring.Metrics.Data.csproj" />
-    <ProjectReference Include="..\SFA.DAS.Payments.Monitoring.Metrics.Domain\SFA.DAS.Payments.Monitoring.Metrics.Domain.csproj" />
+    <Compile Remove="Infrastructure\Configuration\**" />
+    <EmbeddedResource Remove="Infrastructure\Configuration\**" />
+    <None Remove="Infrastructure\Configuration\**" />
```
Comment (tom-gough) [inline src/SFA.DAS.Payments.Monitoring.Metrics.Application/SFA.DAS.Payments.Monitoring.Metrics.Application.csproj:10]: Is this needed/intended as part of this ticket?

**Review by tom-gough — COMMENTED** [review]
Looks good, just one query I've left about a csproj change before approving.

**Review by tom-gough — DISMISSED** [review]

**Review by tom-gough — APPROVED** [review]

**Review by tom-gough — APPROVED** [review]

File: src/SFA.DAS.Payments.Monitoring.Metrics.Data/MetricsQueryDataContext.cs [hunk 2]
```diff
@@ -99,32 +99,44 @@ group by
...
                     group.Key.Ukprn,
                     group.Key.ContractType,
-                    Amount = group.Sum(rp => rp.Amount)
+                    Amount = group.Sum(requiredPaymentInGroup => requiredPaymentInGroup.Amount)
                 })
                 .ToListAsync(cancellationToken)
                 .ConfigureAwait(false);
 
-            return amounts.Select(group => new ProviderContractTypeAmounts()
+            var uniqueUkprns = providerMetrics.Select(x => x.Ukprn).Distinct();
```
Comment (hardik-desai-kcl) [inline src/SFA.DAS.Payments.Monitoring.Metrics.Data/MetricsQueryDataContext.cs:125]: I dont think this line is required, result was already grouped by Ukprn ==> group.Key.Ukprn so return amounts.Select(group => new ProviderContractTypeAmounts() would produce same result as new code

**Review by hardik-desai-kcl — CHANGES_REQUESTED** [review]

File: src/SFA.DAS.Payments.Monitoring.Metrics.Data/MetricsQueryDataContext.cs (hunk 2 above)
Comment (carlbm) [reply src/SFA.DAS.Payments.Monitoring.Metrics.Data/MetricsQueryDataContext.cs:125]: The group key is both Ukprn and Contract type, so if there are both contract types there will be two records

**Review by hardik-desai-kcl — DISMISSED** [review]

**Review by hardik-desai-kcl — APPROVED** [review]

**Pull request merged**